app/packages/looker/src/index.ts

```diff
@@ -642,9 +646,49 @@ export abstract class Looker<
     );
   }
 
+  /**
+   * Some label types (example: segmentation, heatmap) can have their mask datas stored in disks,
+   * we want to impute the `mask` property of those labels with what's stored in the disk
+   */
+  private async imputeOverlayMaskFromDisk(sample: Sample) {
```

Comment:
This seems like something that should be done in a worker thread, no? Maybe this isn't blocking, but it seems like a bottle neck to load all this data into the main thread, only to transfer it.

Reply:
Makes sense. Addressed in next commit.